cfgov/v1/static/js/admin/ask_cfpb_tips.js

```diff
@@ -26,22 +28,94 @@
           buttonset.append(buttonElement);
           
           buttonElement.on('click', function (event) {
+
+
             var sel = window.getSelection();
-            var range = sel.getRangeAt(0).cloneRange();
-            var contents = range.cloneContents();
-            var elem = document.createElement("aside");
-            elem.className = "answer-module";
-            elem.appendChild(contents);
-            range.deleteContents();
-            range.insertNode(elem);
-
-            $(range.commonAncestorContainer).find('*').each(function() {
-              if ($(this).is(':empty') || $(this).text() === '') {
-                $(this).remove();
+            var node = sel.baseNode;
+            
+            var tip = $(node).closest( '.' + baseClass )[0];
+
+            // If selection is currently styled as a tip,
+            // remove the tip wrapper element.
+            if ( tip ) {
+              var docFrag = document.createDocumentFragment();
+              while ( tip.firstChild ) {
+                var child = tip.removeChild( tip.firstChild );
+                docFrag.appendChild(child);
+              }
+              tip.parentNode.replaceChild(docFrag, tip);
+              _this.options.editable.setModified();
+              return;
+            }
+
+            // Otherwise, get the selected range. If it starts or ends
+            // with a text node or inline element (a, strong, i), 
+            // expand the selection either to that node's 
+            // start or end bound, or the start or end bound of its parent
+            // if it's not a direct child of the text input div.
+            var range = rangy.getSelection().getRangeAt(0);
```

Comment:
Would it be possible to use the https://developer.mozilla.org/en-US/docs/Web/API/TreeWalker?